package exception;

import java.util.Random;
import java.util.Scanner;

public class UpDownGame {
	public static void main(String[] args) {
		Random random = new Random();
		int n = random.nextInt(99)+1;
		int you = 0;

		int count = 0;
		try(Scanner sc = new Scanner(System.in)){
			while(true) {
				System.out.print("1부터 99사이의 정수를 입력하세요 > ");
				if(!sc.hasNextInt()) { // 입력이 잘못되었을 때
					System.out.println("값이 잘못 입력되었습니다.");
					sc.nextLine();
					continue;
				}
				you = sc.nextInt();
				sc.nextLine();
				if(!(1<= you && you <= 99)) {
					System.out.println("값의 범위가 틀렸습니다. 1<= x <= 99 사이의 정수로 입력하세요.");

				}else if(you != n) {
					System.out.println("입력값은 : "+ you +"입니다.");
					if(you < n) {
						System.out.println("UP@");
					}else {
						System.out.println("Down@!");
					}
					count++;
				}else {
					System.out.println("맞았습니다. 컴퓨터가 생각한 값은 "+ n + "입니다");
					System.out.println("count : "+ count);
					switch(count){
					case 1,2,3,4 : System.out.println("GREAT!"); break;
					case 5,6,7,8 : System.out.println("Good!"); break;
					case 9,10,11,12 : System.out.println("그럭저럭"); break;
					case 13,14,15,16 : System.out.println("아쉽"); break;
					default : System.out.println("분발하세요"); break;
					}
					break;
				}

			}
		}
		System.out.println("종료");
	}
}
